tests/unit/dataactvalidator/test_c21_award_financial.py

```diff
@@ -0,0 +1,327 @@
+        ussgl487100_downward_adjus_cpe=ussgl487100_downward_adjus_cpe,
+        ussgl497100_downward_adjus_cpe=ussgl497100_downward_adjus_cpe,
+        ussgl487200_downward_adjus_cpe=ussgl487200_downward_adjus_cpe,
+        ussgl497200_downward_adjus_cpe=ussgl497200_downward_adjus_cpe,
+        deobligations_recov_by_awa_cpe=deobligations_recov_by_awa_cpe,
+        tas=tas,
+        program_activity_code=program_activity_code,
+        submission_id=submission_id
+    )
+
+    af2 = AwardFinancialFactory(
+        ussgl480100_undelivered_or_fyb=ussgl480100_undelivered_or_fyb,
+        ussgl480100_undelivered_or_cpe=ussgl480100_undelivered_or_cpe,
+        ussgl483100_undelivered_or_cpe=ussgl483100_undelivered_or_cpe,
+        ussgl488100_upward_adjustm_cpe=ussgl488100_upward_adjustm_cpe,
+        obligations_undelivered_or_fyb=obligations_undelivered_or_fyb,
+        obligations_undelivered_or_cpe=obligations_undelivered_or_cpe,
+        ussgl490100_delivered_orde_fyb=ussgl490100_delivered_orde_fyb,
+        ussgl490100_delivered_orde_cpe=ussgl490100_delivered_orde_cpe,
+        ussgl493100_delivered_orde_cpe=ussgl493100_delivered_orde_cpe,
+        ussgl498100_upward_adjustm_cpe=ussgl498100_upward_adjustm_cpe,
+        obligations_delivered_orde_fyb=obligations_delivered_orde_fyb,
+        obligations_delivered_orde_cpe=obligations_delivered_orde_cpe,
+        ussgl480200_undelivered_or_fyb=ussgl480200_undelivered_or_fyb,
+        ussgl480200_undelivered_or_cpe=ussgl480200_undelivered_or_cpe,
+        ussgl483200_undelivered_or_cpe=ussgl483200_undelivered_or_cpe,
+        ussgl488200_upward_adjustm_cpe=ussgl488200_upward_adjustm_cpe,
+        gross_outlays_undelivered_fyb=gross_outlays_undelivered_fyb,
+        gross_outlays_undelivered_cpe=gross_outlays_undelivered_cpe,
+        ussgl490200_delivered_orde_cpe=ussgl490200_delivered_orde_cpe,
+        ussgl490800_authority_outl_fyb=ussgl490800_authority_outl_fyb,
+        ussgl490800_authority_outl_cpe=ussgl490800_authority_outl_cpe,
+        ussgl498200_upward_adjustm_cpe=ussgl498200_upward_adjustm_cpe,
+        gross_outlays_delivered_or_fyb=gross_outlays_delivered_or_fyb,
+        gross_outlays_delivered_or_cpe=gross_outlays_delivered_or_cpe,
+        gross_outlay_amount_by_awa_fyb=gross_outlay_amount_by_awa_fyb,
+        gross_outlay_amount_by_awa_cpe=gross_outlay_amount_by_awa_cpe,
+        obligations_incurred_byawa_cpe=obligations_incurred_byawa_cpe,
+        ussgl487100_downward_adjus_cpe=ussgl487100_downward_adjus_cpe,
+        ussgl497100_downward_adjus_cpe=ussgl497100_downward_adjus_cpe,
+        ussgl487200_downward_adjus_cpe=ussgl487200_downward_adjus_cpe,
+        ussgl497200_downward_adjus_cpe=ussgl497200_downward_adjus_cpe,
+        deobligations_recov_by_awa_cpe=deobligations_recov_by_awa_cpe,
+        tas=tas,
+        program_activity_code=program_activity_code,
+        submission_id=submission_id
+    )
+
+    op = ObjectClassProgramActivityFactory(
```

Comment:
Same question as the one I had for the `test_failure` method in the C20 unit test--should we make sure the `<` part of the rule is tested along with the `=` part.

I think the broader question, however, is: if any one of the comparisons fails, the entire rule fails. Is this really what we want (will say more in the PR comments).